graph/patterns/dictionary.md

```diff
@@ -103,34 +115,43 @@ PATCH /item/userTags
 }
 ```
 
-### Declaring a complex typed dictionary
+### Complex typed dictionary
+
+#### CSDL declaration
 Dictionaries can also contain complex types whose values may be constrained to a particular set of complex types.
 
 The following example defines a complex type **roleSettings**, an **assignedRoleGroupDictionary** that contains **roleSettings**, and an **assignedRoles** property that uses the dictionary..
 
 ```xml
-<EntityType Name="principal">
-  ...
-  <Property Name="assignedRoles" Type="self.assignedRoleGroupDictionary">
-</EntityType>
-
-<ComplexType Name="roleSettings">
-  <Property Name ="domain" Type="Edm.String" Nullable="false" />
-</ComplexType>
-
-<ComplexType Name="assignedRoleGroupDictionary" BaseType="graph.Dictionary">
-  <!-- Note: Strongly-typed dictionary
-  of roleSettings
-  keyed by name of roleGroup. -->
-  <Annotation Term="Org.OData.Validation.V1.OpenPropertyTypeConstraint">
-    <Collection>
-      <String>microsoft.graph.roleSettings</String>
-    </Collection>
-  </Annotation>
-</ComplexType>
+<Schema Namespace="microsoft.graph"> <!--NOTE: the namespace that declares the Dictionary complex type *must* be microsoft.graph-->
+  <ComplexType Name="Dictionary" OpenType="true">
+    <Annotation Term="Core.Description" String="A dictionary of name-value pairs. Names must be valid property names, values may be restricted to a list of types via an annotation with term `Validation.OpenPropertyTypeConstraint`." />
+  </ComplexType>
+</Schema>
+<Schema Namespace="WorkloadNamespace">
+  <EntityType Name="principal">
```

Comment:
nit: consider moving this to *after* the definition of assignedRoleGroupDictionary, so that the reader sees the progression (the complex type, the dictionary of complex type, and then the use of the dictionary)